src/torchmetrics/image/vif.py

```diff
@@ -0,0 +1,67 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.functional.image.vif import _vif_per_channel
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.distributed import reduce
+
+
+class VisualInformationFidelity(Metric):
+    """Compute Pixel Based Visual Information Fidelity (VIF_).
+
+    As input to ``forward`` and ``update`` the metric accepts the following input
+
+    - ``preds`` (:class:`~torch.Tensor`): Predictions from model of shape ``(N,C,H,W)``
+    - ``target`` (:class:`~torch.Tensor`): Ground truth values of shape ``(N,C,H,W)``
+
+    As output of `forward` and `compute` the metric returns the following output
+
+    - ``vif-p`` (:class:`~torch.Tensor`): Tensor with vif-p score
+
+    Args:
+        sigma_n_sq: variance of the visual noise
+        kwargs: Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+    """
+
+    is_differentiable = True
+    higher_is_better = True
+    full_state_update = False
+
+    vif_score: Tensor
+    total: Tensor
+
+    def __init__(self, sigma_n_sq: float = 2.0, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+        self.add_state("vif_score", default=tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("total", default=tensor(0.0), dist_reduce_fx="sum")
+        self.sigma_n_sq = sigma_n_sq
```

Comment:
lets add some input validation similar to:
https://github.com/Lightning-AI/torchmetrics/blob/ef1af350befa32fe0ad4bf9e91ec544ee0dc5e66/src/torchmetrics/image/rase.py#L77-L78

Reply:
Added